### ISSUE TITLE
Django Admin - Adapt sidenav to be an overlay/slideout that doesn't reflow the page

### ISSUE THREAD

**Issue by reitermb** [raft-tech/TANF-app#1411]
**Description:**
 **As a django admin user, I want to access the sidenav at all breakpoints.** Currently, no visible focus indicator is present for the nav toggle button in either the open or closed state (see screenshots)
![image](https://user-images.githubusercontent.com/84722778/138108693-2e27f90b-1efd-4990-9b74-daf607b0daca.png)
![image](https://user-images.githubusercontent.com/84722778/138108711-b51bf6be-5d66-4b68-a677-2b2122b57179.png)

**Acceptance Criteria:** 
- [ ] Sidenav must not disappear or fully block screen content at zoom/breakpoint(s) described in comments below.
- [ ] Sidenav open/close button has a text label, not just an icon
- [ ] Sidenav open/close button conveys open/closed state when read by screenreaders (aria expanded = true/false)
- [ ] Focus is appropriately defaulted to the close button when the sidenav is opened
- [ ] Focus is appropriately trapped within the sidenav when it is opened
- [ ] Sidenav is not tabbable/readable without opening
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
_Create a list of granular, specific work items that must be completed to deliver the desired outcomes of this issue_ 

- [ ] Change left side nav button to text
- [ ] Move the Close form of button to top left of element
- [ ] Stop main element from shifting when menu pops out, and darken
- [ ] Run Testing Checklist and confirm all tests pass

**Supporting Documentation:**
[HackMD notes](https://hackmd.io/3zR7vj7GQ7uidJmLR-eCYw?view#Follow-On-Stuff)

![image](https://user-images.githubusercontent.com/28515874/140802045-0411625c-32c5-49ab-9f84-4becb94d2350.png)
Mockup of target state side nav (open/closed)

**Comment by abottoms-coder** [comment]
@reitermb 

>  1. Sidenav acts as an overlay on top of page content rather than reflowing page content
>  2. Sidenav is not tabbable/readable without opening

For item 1 here, this would require a larger burden of effort. Can we still be compliant without this piece? To ask another way, is this a nice-to-have or absolutely required?

For item 2, does this mean the contents are not tabbable without opening? Presumably we would want the opener arrows tabbable. Just a minor clarifying question.

cc @riatzukiza @jorgegonzalez

**Comment by reitermb** [comment]
Clarifying scope: 

The core goal here is to move from the **current state** where: 
The sidenav is hidden at a certain window size / zoom level

to the **target state**:
100% of page content (including the sidenav) is viewable up to 200% zoom—including at smaller window sizes/low screen resolutions like 1280 X 768.

Reflowing the main content area instead of overlaying the sidenav _is_ a feasible approach here as long as in compressing the rest of the content we don't cause any of it to flow offscreen (One example being that tables would need to be scrollable since we'd probably be _very_ limited on the number of visible columns in a high zoom / low resolution / sidenav open scenario).

cc: @riatzukiza @jorgegonzalez @abottoms-coder

**Comment by riatzukiza** [comment]
Already first day into this, the button doesn't work in our test website for django 508. I have the script from the admin panel source copied in, and have been able to make some changes. I can't see what is wrong

**Comment by riatzukiza** [comment]
Started a PR for this into DA508 https://github.com/raft-tech/django-admin-508/pull/16

**Comment by riatzukiza** [comment]
![image](https://user-images.githubusercontent.com/10676925/157332479-782481f3-9ebe-49be-96cd-b4923c01d378.png)
![image](https://user-images.githubusercontent.com/10676925/157332508-a13a6837-fa16-4efb-b705-309cff922546.png)
![image](https://user-images.githubusercontent.com/10676925/157332543-c311aae4-6027-4410-9d5c-7c773d01d00a.png)

**Comment by riatzukiza** [comment]
Waiting for review in django admin 508 repo to introduce the new version here.
